git-hooks/pre-commit

```diff
@@ -6,35 +6,65 @@ const fs = require('fs');
 const execa = require('execa');
 const chalk = require('chalk');
 
+const fixAndStage = (jsFiles, localDir) =>
+    execa('eslint', ['--fix', ...jsFiles], {
+        stdio: 'inherit',
+        localDir,
+    }).then(() =>
+        execa('git', ['add', ...jsFiles], {
+            stdio: 'inherit',
+        })
+    );
+
+const fixStagedIn = (stagedFiles, location) => {
+    const locations = {
+        static: {
+            filter: file => file.startsWith('static') && file.endsWith('.js'),
+            localDir: process.cwd(),
+        },
+        ui: {
+            filter: file =>
+                file.startsWith('ui') &&
+                (file.endsWith('.js') || file.endsWith('.jsx')),
+            localDir: 'ui',
+        },
+    };
+    const jsFiles = stagedFiles
+        .filter(locations[location].filter)
+        .filter(file => fs.existsSync(file));
+
+    if (jsFiles.length) {
+        return fixAndStage(jsFiles, locations[location].localDir).then(
+            () => stagedFiles
+        );
+    }
+
+    return Promise.resolve(stagedFiles);
+};
+
 execa
     .stdout('git', ['diff', '--name-only', '--cached', '--diff-filter=AM'])
     .then(staged => staged.split('\n'))
     // lint and commit any lint fixes
-    .then(staged => {
-        const jsFiles = staged
-            .filter(file => file.startsWith('static') && file.endsWith('.js'))
-            .filter(file => fs.existsSync(file));
-        if (jsFiles.length) {
-            return execa('eslint', ['--fix', ...jsFiles], {
-                stdio: 'inherit',
-            })
-                .then(() =>
-                    execa('git', ['add', ...jsFiles], {
-                        stdio: 'inherit',
-                    })
-                )
-                .then(() => staged);
-        }
-        return staged;
-    })
-    // make sure yarn.lock is commited
+    .then(staged =>
+        fixStagedIn(staged, 'static').then(() => fixStagedIn(staged, 'ui'))
```

Comment:
don't we just need the list of files, then pass that whole thing to eslint to manage? don't need 2 tasks to do it?